summoner-cli/src/Summoner/Source.hs

```diff
@@ -33,18 +34,22 @@ showSource = \case
     File _ -> "File"
     Link _ -> "Link"
 
+-- TODO: return Maybe
 matchUrl :: Source -> Either TomlBiMapError Text
 matchUrl (Url url) = Right url
 matchUrl e         = Left $ WrongConstructor "Url" $ showSource e
 
+-- TODO: return Maybe
 matchFile :: Source -> Either TomlBiMapError FilePath
 matchFile (File file) = Right file
 matchFile e           = Left $ WrongConstructor "File" $ showSource e
 
+-- TODO: return Maybe
 matchLink :: Source -> Either TomlBiMapError Text
 matchLink (Link link) = Right link
 matchLink e           = Left $ WrongConstructor "Link" $ showSource e
 
+-- DEPRECATED: To be removed in 2.0
```

Comment:
Since we're still using this function, we will see `DEPRECATED` warning each time we build a project. I just thought that it could be annoying that's why I haven't added a pragma. But I'm also fine with adding it. Do you think it will help us not to forget to remove this function?

Reply:
Mm, I see. Yeah, probably not worth it. But proper haddock is definitely needed :slightly_smiling_face: